overlord/servicestate/quota_control.go

```diff
@@ -51,26 +54,37 @@ func checkSystemdVersion() {
 
 func init() {
 	checkSystemdVersion()
-	checkMemoryCGroupEnabled()
+	refreshCGroupControllerStatus()
 }
 
-func checkMemoryCGroupEnabled() {
-	memoryCGroupError = memoryCGroupEnabled()
+// refreshCGroupControllerStatus refreshes status for varius cgroup controller features
+// as they need to be activated to use the individual quota groups. The features
+// we check for are cpuacct(v1)/cpu(v2), cpuset and memory.
+func refreshCGroupControllerStatus() {
+	cpuControllerName := "cpuacct"
+	cgroupVersion, err := cgroup.Version()
+	if err != nil && cgroupVersion == 2 {
+		cpuControllerName = "cpu"
+	}
+
+	memoryCGroupError = checkCGroupControllerEnabled("memory")
+	cpuCGroupError = checkCGroupControllerEnabled(cpuControllerName)
+	cpuSetCGroupError = checkCGroupControllerEnabled("cpuset")
 }
```

Comment:
this needs to be adjusted given what we learned in https://github.com/snapcore/snapd/pull/11339

I would suggest syncing with @atesburak about the direction here, but for the time being for your PR here, we can simply ignore the fact that a cgroup may be disabled and live with the poor user experience, since confusing/unhelpful error messages are better than broken devices

Reply:
marking as blocked until this code is removed to prevent re-introducing this regression here

Reply:
Hi Ian, I have also changed the logic of how AllQuotas work - can we possibly rendezvous about this monday?

Lets just wait throwing all of this logic out.

Reply:
sure we can discuss after the SU (either tomorrow or Monday)